soccer/src/soccer/TeamInfo.hpp

```diff
@@ -4,33 +4,36 @@
 
 #include <iostream>
 #include <rj_common/time.hpp>
+#include <rj_msgs/msg/team_info.hpp>
 #include <string>
 #include <vector>
 
 // Information about a single team.
 class TeamInfo {
 public:
+    using Msg = rj_msgs::msg::TeamInfo;
+
     // The team's name (empty string if operator has not typed anything).
-    std::string name = "";
+    std::string name;
     // The number of goals scored by the team during normal play and overtime.
-    uint score = 0;
+    int score = 0;
```

Comment:
hm, can score be negative?

Reply:
Not In Real Life (:tm:), but I've found that unsigned integers are a pretty poor way to express the constraint of "nonnegative integer" (at least, not without overflow/underflow checking). I usually use them for memory/size-related variables and otherwise avoid using them when there's a choice.

The reason is that `uint` _implicitly_ captures the constraint while giving the illusion that it's capturing it _explicitly_. If there's any chance I'd subtract one thing from another - something like:
```cpp
const auto goal_differential = our_info.score - their_info.score;
if (goal_differential > SOME_POSITIVE_NUMBER) { /* be conservative */ }
if (goal_differential < SOME_NEGATIVE_NUMBER) { /* be aggressive */ }
```
then it's just easier to save the headache down the line and make it a regular int, assuming we don't need the range (and I'm pretty sure mercy rule kicks in before one team scores 2 billion goals :wink: )

---

TL;DR: I am a fool; I've fallen prey to one of the classic blunders!
```cpp
for (size_t i = N; i >= 0; i--) { ... }
```